jujugui/static/gui/src/app/app.js

```diff
@@ -111,44 +111,45 @@ YUI.add('juju-gui', function(Y) {
         target: '#shortcut-settings',
         toggle: true,
         callback: function(evt, target) {
-          const shortcutHelp = Y.one('#shortcut-help');
+          const shortcutHelp = document.querySelector('#shortcut-help');
           if (shortcutHelp) {
-            shortcutHelp.hide();
+            shortcutHelp.classList.add('hidden');
           }
 
-          if (target && !target.getHTML().length) {
+          if (target && !target.children.length) {
             ReactDOM.render(
               <window.juju.components.Settings
                 disableCookie={localStorage.getItem('disable-cookie')}
                 disableAutoPlace={localStorage.getItem('disable-auto-place')}
                 forceContainers={localStorage.getItem('force-containers')} />,
-              target.getDOMNode());
+              ReactDOM.findDOMNode(target));
 
             // This is only added to the DOM once and is checked if it exists
             // above. It's hidden and then shown, so this event is not auto
             // cleaned up, but can only occur once.
-            target.one('#save-settings').on('click', function(ev) {
-              var fields = target.all('input');
-              fields.each(function(node) {
-                // If it's a checkbox:
-                if (node.get('type') === 'checkbox') {
-                  // and if it's checked set that value to localStorage.
-                  if (node.get('checked')) {
-                    localStorage.setItem(node.getAttribute('name'), true);
+            target.querySelector('#save-settings').addEventListener(
+              'click', ev => {
+                const fields = target.querySelectorAll('input');
+                fields.each(function(node) {
+                  // If it's a checkbox:
+                  if (node.get('type') === 'checkbox') {
+                    // and if it's checked set that value to localStorage.
+                    if (node.get('checked')) {
+                      localStorage.setItem(node.getAttribute('name'), true);
+                    } else {
+                      // otherwise unset it from the localStorage.
+                      localStorage.removeItem(node.getAttribute('name'));
+                    }
                   } else {
-                    // otherwise unset it from the localStorage.
-                    localStorage.removeItem(node.getAttribute('name'));
+                    localStorage.setItem(
```

Comment:
It looks like the diff gets really wacky here, but all I did was indent the code inside the event listener.